### PR TITLE
refactor(odyssey-react): use Text for List

### PR DESCRIPTION
**Pull request by jeffbelser-okta**

### CODE REVIEW

File: packages/odyssey-react/src/components/List/DescriptionDetails.tsx [hunk 1]
```diff
@@ -28,7 +29,7 @@ const DescriptionDetails = forwardRef<HTMLElement, DescriptionDetailsProps>(
 
     return (
       <dd {...omitProps} ref={ref} className={styles.details}>
-        {children}
+        <Text children={children} />
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/List/DescriptionDetails.tsx:32]: I considered changing the type annotation here to only allow `ReactText` children but decided against a breaking change (or the complexity to support both APIs at once) and left the API open to composition if necessary

File: packages/odyssey-react/src/components/List/List.test.tsx [hunk 2]
```diff
@@ -111,7 +111,7 @@ describe("DescriptionTerm", () => {
     render(<List.Term ref={ref} children="term" />);
 
     expect(ref).toHaveBeenCalledTimes(1);
-    expect(ref).toHaveBeenLastCalledWith(screen.getByText("term"));
+    expect(ref).toHaveBeenLastCalledWith(screen.getByText("term").parentNode);
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/List/List.test.tsx:114]: 🙄

**Comment by edburyenegren-okta** [comment]
Similar to the convos going on in Heading and Button, why not `<Text as="dd">` and `<Text as="dt">` rather than inserting an element to style a leaf?

That way, the plain element styles live together in `Text` and can be applied to non-react `dl`s as they are with `abbr`, `address`, and so on rather than styling being split across `List.module.scss` and `Text.module.scss` for an HTML element.

**Comment by jeffbelser-okta** [comment]
@edburyenegren-okta That discussion is def one we should have soon - sounds like we need someone to lead that effort to help drive us to a decision. From my perspective, it comes down to what type of API we want `<Text />` to have, Text as dt and dd feels a bit strange at first glance but would love to hear out all sides

**Review by andrewberg-okta — COMMENTED** [review]
Without adding a `<Text>` wrapper to `<List>` this component can't stand alone and have Odyssey styles. Otherwise we're asking any dowsteam team to wrap all their list content to get propper styles.

**Comment by jeffbelser-okta** [comment]
> Without adding a `<Text>` wrapper to `<List>` this component can't stand alone and have Odyssey styles. Otherwise we're asking any dowsteam team to wrap all their list content to get propper styles.
> 
> Specifically without that wrapper the list item markers will get page styling eg: https://codepen.io/andrewberg-okta/pen/wvqERMW?editors=1100

This is a good point. Great fodder for our API discussion too. My immediate reaction is that some amount of composition for text when `children` is typed as `ReactNode` is OK. If children was typed as `ReactText` then the assumption might be that we already provide an intermediate `Text` component as part of our List implementation to take care of this

**Review by andrewberg-okta — APPROVED** [review]

**Pull request merged**